In [2]:
import tensorflow as tf
import numpy as np
import pandas as pd

#### Data creation

In [4]:
T, F = 1., -1.
bias = 1.

In [32]:
train_in = [
    [T, T, bias],
    [T, F, bias],
    [F, T, bias],
    [F, F, bias]
]

train_out = [
    [T],
    [F],
    [F],
    [F]
]

#Init = tf.random_normal([3, 1])
w = tf.Variable(tf.random_normal([3, 1]))

mult = tf.matmul(train_in, w)
#cost = tf.reduce_sum(mult)

actFun = tf.tanh(mult)
error = tf.subtract(train_out, actFun)
mse = tf.reduce_mean(tf.square(error))

delta = tf.matmul(train_in, error, transpose_a=True)
train = tf.assign(w, tf.add(w, delta))

sess = tf.Session()
sess.run(tf.initialize_all_variables())

err, target = 1, 0
epoch, max_epoch = 0, 20

while err>target and epoch < max_epoch:
    epoch += 1
    err, _ = sess.run([mse, train])
    print('Epoch:', epoch, 'mse:',err)

Epoch: 1 mse: 1.1156948
Epoch: 2 mse: 0.23107085
Epoch: 3 mse: 0.0043377667
Epoch: 4 mse: 0.0014718625
Epoch: 5 mse: 0.0009494263
Epoch: 6 mse: 0.00071433594
Epoch: 7 mse: 0.00057790143
Epoch: 8 mse: 0.0004880072
Epoch: 9 mse: 0.00042384077
Epoch: 10 mse: 0.00037533208
Epoch: 11 mse: 0.00033704937
Epoch: 12 mse: 0.00030581275
Epoch: 13 mse: 0.00027966217
Epoch: 14 mse: 0.0002573353
Epoch: 15 mse: 0.0002379681
Epoch: 16 mse: 0.00022096965
Epoch: 17 mse: 0.00020589861
Epoch: 18 mse: 0.00019243537
Epoch: 19 mse: 0.00018032995
Epoch: 20 mse: 0.00016938918


#### MNIST

In [33]:
from tensorflow.examples.tutorials.mnist import input_data

In [36]:
mnist = input_data.read_data_sets('MNIST_data', one_hot = True)

Extracting MNIST_data\train-images-idx3-ubyte.gz
Extracting MNIST_data\train-labels-idx1-ubyte.gz
Instructions for updating:
Please use tf.one_hot on tensors.
Extracting MNIST_data\t10k-images-idx3-ubyte.gz
Extracting MNIST_data\t10k-labels-idx1-ubyte.gz


In [40]:
train_x = mnist.train.images
print(train_x.shape)
train_y = mnist.train.labels
print(train_y.shape)
test_x = mnist.test.images
print(test_x.shape)
test_y =mnist.test.labels
print(test_y.shape)

(55000, 784)
(55000, 10)
(10000, 784)
(10000, 10)


### Model with
- 1 Input layer ((55000, 784))
- 1 hidden layer with 500 units (())
- 1 output layer with i unit

In [ ]:
## Input to hidden layer1 weights and bias

W01 = tf.Variable(tf.random_normal([500, train_x.shape[1]]))
B01 = tf.Variable(tf.random_normal)
